### PR TITLE
chore(ingestion): have populateTeamDataStep log team resolution mismatches

### PR DESCRIPTION
**Pull request by xvello**
## Problem

Follow-up to https://github.com/PostHog/posthog/pull/14341, we do see some points on the `ingestion_team_resolution_checks` counter, indicating mismatches between the team_id
set by capture and the one resolved by plugin-server.

image

## Changes

Log these occurrences so that we can determine what is happening. Now that we have confirmed it's a few dozen cases a day, we can be verbose about them.

### CODE REVIEW

File: plugin-server/src/worker/ingestion/event-pipeline/populateTeamDataStep.ts [hunk 1]
```diff
@@ -53,6 +54,13 @@ export async function populateTeamDataStep(
             teamResolutionChecksCounter.labels({ check_ok: checkOk }).inc()
             // statsd copy as prometheus is currently not supported in worker threads.
             runner.hub.statsd?.increment('ingestion_team_resolution_checks', { check_ok: checkOk })
+            if (!checkOk) {
```
Comment (tomasfarias) [inline plugin-server/src/worker/ingestion/event-pipeline/populateTeamDataStep.ts:57]: I am very confused by js but: 

```js
Welcome to Node.js v18.14.0.
Type ".help" for more information.
> const checkOk = 'false'
undefined
> !checkOk
false
```

I think we should do a different check here.

Comment (xvello) [reply plugin-server/src/worker/ingestion/event-pipeline/populateTeamDataStep.ts:57]: oh, good catch! thanks!

Comment (tomasfarias) [reply plugin-server/src/worker/ingestion/event-pipeline/populateTeamDataStep.ts:57]: I guess it's consistent:

```js
> const checkOk = 'true'
undefined
> !checkOk
false
```

Comment (xvello) [reply plugin-server/src/worker/ingestion/event-pipeline/populateTeamDataStep.ts:57]: Yup, that's a "truthie" check, non-empty string is `truthy`

File: plugin-server/src/worker/ingestion/event-pipeline/populateTeamDataStep.ts [hunk 2]
```diff
@@ -49,7 +50,17 @@ export async function populateTeamDataStep(
     if (event.team_id) {
         if (event.token) {
             const team = await runner.hub.teamManager.getTeamByToken(event.token)
-            const checkOk = team?.id === event.team_id ? 'true' : 'false'
```
Comment (tomasfarias) [inline plugin-server/src/worker/ingestion/event-pipeline/populateTeamDataStep.ts:52]: Thanks, I was not a fan of this line 😅

**Review by tomasfarias — APPROVED** [review]
Let's see what's happening, looks really surprising 👀

**Pull request merged**